### PR TITLE
fix: remove extra term() argument at start of macrocallback spec

### PR DESCRIPTION
**Pull request by Eiji7**
related: #755 and #756
ping @josevalim

### CODE REVIEW

File: lib/ex_doc/language/elixir.ex [hunk 1]
```diff
@@ -109,11 +109,14 @@ defmodule ExDoc.Language.Elixir do
       if actual_def in module_data.private.optional_callbacks, do: ["optional"], else: []
 
     specs =
-      case Map.fetch(module_data.private.callbacks, actual_def) do
-        {:ok, specs} ->
+      case {kind, Map.fetch(module_data.private.callbacks, actual_def)} do
+        {:macrocallback, {:ok, specs}} ->
+          Enum.map(specs, &remove_callback_term/1)
+
+        {:callback, {:ok, specs}} ->
```
Comment (wojtekmach) [inline lib/ex_doc/language/elixir.ex:116]:
```suggestion
      case Map.fetch(module_data.private.callbacks, actual_def) do
        if kind == :macrocallback do
          Enum.map(specs, &remove_callback_term/1)
        else
          specs
        end
```

**Review by wojtekmach — COMMENTED** [review]
Nice catch!

File: lib/ex_doc/language/elixir.ex (hunk 1 above)
Comment (Eiji7) [reply lib/ex_doc/language/elixir.ex:116]: @wojtekmach I was also curious about "preferred style" in core team. That was also my first version, but I was thinkining that I'm adding unnecessary indention, so I took a look at other `case` in this file …

I found similar code (variable + function call in tuple) in lines 770 and 835 (or 759 and 824 before change). Before commit those I would like to ask about consistency with mentioned lines if you don't mind. :slightly_smiling_face:

File: lib/ex_doc/language/elixir.ex [hunk 2]
```diff
@@ -109,11 +109,14 @@ defmodule ExDoc.Language.Elixir do
       if actual_def in module_data.private.optional_callbacks, do: ["optional"], else: []
 
     specs =
-      case Map.fetch(module_data.private.callbacks, actual_def) do
-        {:ok, specs} ->
+      case module_data.private.callbacks do
+        %{^actual_def => specs} when kind == :macro_callback ->
```
Comment (josevalim) [inline lib/ex_doc/language/elixir.ex:113]:
```suggestion
        %{^actual_def => specs} when kind == :macrocallback ->
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

File: lib/ex_doc/language/elixir.ex (hunk 1 above)
Comment (Eiji7) [reply lib/ex_doc/language/elixir.ex]: José Valim already fixed that much better with `when`. :+1: :heart:

**Pull request merged**